### PR TITLE
Refactor blocks

### PR DESCRIPTION
**Pull request by zatan**

### CODE REVIEW

File: glitter/blockadmin/blocks.py [hunk 1]
```diff
@@ -36,7 +37,11 @@ def __init__(self, *args, **kwargs):
     # Use the block admin class by default
     def register(self, model_or_iterable, admin_class=None, **options):
         if not admin_class:
-            admin_class = BlockModelAdmin
+            admin_class = BlockAdmin
+
+        category = options.pop('category', None)
+        if category:
+            self.register_block(model_or_iterable, category)
```
Comment (tomkins) [inline glitter/blockadmin/blocks.py]: `register_block` can't currently take an iterable? Though that should be a nice easy one to fix.

**Comment by tomkins** [comment]
`block_admin.py` has been moved/renamed, but there's nothing left in for backwards compatibility.

Would be nice if there's a `DeprecationWarning` in block_admin.py with some backwards compatbility so we can avoid a disruptive upgrade. So we need to be able to use the old class names (eg. `BlockAdminSite`) for a bit longer.

Loving the refactor!

**Comment by zatan** [comment]
Updated

**Comment by tomkins** [comment]
There's a few faulty imports:

```
glitter/blocks/form/admin.py:from .block_admin import blocks
glitter/blocks/latest_tweets/admin.py:from .block_admin import blocks
glitter/blocks/related_pages/admin.py:from .block_admin import blocks
glitter/blocks/video/admin.py:from .block_admin import blocks
```

Also need block_admin.py to keep a few more classes for backwards compatibility, (StackedInline, TabularInline).

Happy with the rest though!

**Comment by tomkins** [comment]
Still need these updated:

```
glitter/blocks/latest_tweets/admin.py:from .block_admin import blocks
glitter/blocks/related_pages/admin.py:from .block_admin import blocks
```

**Comment by tomkins** [comment]
Happy with all the existing stuff, although adding the tests to redactor has now caused tox/jenkins failures.

**Pull request merged**